xarray/core/variable.py

```diff
@@ -1069,6 +1069,32 @@ def chunk(self, chunks={}, name=None, lock=False):
 
         return self._replace(data=data)
 
+    def to_numpy(self) -> np.ndarray:
+        """Coerces wrapped data to numpy and returns a numpy.ndarray"""
+        # TODO an entrypoint so array libraries can choose coercion method?
+        data = self.data
+        try:
+            data = data.to_numpy()
+        except AttributeError:
```

Comment:
Isn't this always failing? Should we avoid it until such a protocol actually exists?

Reply:
That would be my inclination, too.

I can also imagine some other library implementing `to_numpy()` in a weird way, which could cause issues. At the least, there should be a defensive check of `isinstance(data, np.ndarray)`  afterwards.